server/utils/helpers/checkSubcategoryHelper.js

```diff
@@ -0,0 +1,25 @@
+import models from '../../database/models';
+
+const { Subcategory } = models;
+
+const checkSubcategory = (req, res, next) => {
+  // Checking if subcategory exists in the database
+  const subcategoryName = req.body.article.subcategory;
+  if (subcategoryName && subcategoryName.trim()) {
```

Comment:
`subcategoryName && subcategoryName.trim()` this checks if the user entered a subcategory (i.e a subcategory actually exists) and this also checks that it is not a blank space. If the two checks are true then it goes ahead to search if the subcategory exists in the database

Reply:
@JCanaks so why can't you just do only `if (subcategoryName.trim())` ?

Reply:
that way you know it'll always check for blank spaces and also a valid values at the same time

Reply:
There is an error if i check only `if (subcategoryName.trim())` because if `subcategoryName` does not exist it flags an error saying cannot `perform .trim() on undefined` hence the need for the two checks.

Reply:
```
const { subcategory = '' } = req.body.article.subcategory; // and you better change this to subCategory;
if(subcategory.trim())
```
@JCanaks what about now ?

Reply:
subcategory is one word that's why I used all lowercases or would it be preferable to use `subCategory`?

Reply:
```const { subcategory = '' } = req.body.article.subcategory; // and you better change this to subCategory;
if(subcategory.trim())```
@rhenshaw56 It assigns an empty to subcategory even if a subcategory is specified in the req body